e2e/support/test_support.go

```diff
@@ -1136,6 +1209,28 @@ func BindKameletTo(ns, name, from string, to corev1.ObjectReference, properties
 	}
 }
 
+func BindKameletToWithErrorHandler(ns, name, from string, to corev1.ObjectReference, sourceProperties map[string]string, sinkProperties map[string]string, errorHandler map[string]interface{}) func() error {
+	return func() error {
+		kb := v1alpha1.NewKameletBinding(ns, name)
+		kb.Spec = v1alpha1.KameletBindingSpec{
+				Source: v1alpha1.Endpoint{
+					Ref: &corev1.ObjectReference{
```

Comment:
I'd use the same `corev1.ObjectReference` mechanics you're making with the `to` input parameters instead, leaving the declaration of the `from` to the caller. I think you took it the implementation from `BindKameletTo()` func. Check if you can create a func which is returning a general `KameletBinding` and the refactor those methods using it. Always good if we can simplify our code without adding duplicates.